freqtrade/freqtradebot.py

```diff
@@ -690,46 +690,77 @@ def handle_stoploss_on_exchange(self, trade: Trade) -> bool:
+                if trade.stoploss_order_id else None
+        except DependencyException as exception:
+            logger.warning('Unable to fetch stoploss order: %s', exception)
+
+        # If trade open order id does not exist,
+        # it means buy order is fulfilled
+        buy_order_fulfilled = not trade.open_order_id
 
-                stop_price = trade.open_rate * (1 + stoploss)
+        # Limit price threshold
+        # This is the limit price percentage below which you don't want to sell
+        # 0.99 is arbitrary here.
+        # As limit price should always be below price
+        limit_price_pct = 0.99
+
+        # If buy order is fulfilled but there is no stoploss,
+        # then we add a stoploss on exchange
+        if (buy_order_fulfilled and not stoploss_order):
+            if self.edge:
+                stoploss = self.edge.stoploss(pair=trade.pair)
+            else:
+                stoploss = self.strategy.stoploss
 
-                # limit price should be less than stop price.
-                # 0.99 is arbitrary here.
-                limit_price = stop_price * 0.99
+            stop_price = trade.open_rate * (1 + stoploss)
 
+            # limit price should be less than stop price.
+            limit_price = stop_price * limit_price_pct
+
+            try:
                 stoploss_order_id = self.exchange.stoploss_limit(
                     pair=trade.pair, amount=trade.amount, stop_price=stop_price, rate=limit_price
                 )['id']
                 trade.stoploss_order_id = str(stoploss_order_id)
                 trade.stoploss_last_update = datetime.now()
+                return False
 
-            # Or the trade open and there is already a stoploss on exchange.
-            # so we check if it is hit ...
-            elif trade.stoploss_order_id:
-                logger.debug('Handling stoploss on exchange %s ...', trade)
-                order = self.exchange.get_order(trade.stoploss_order_id, trade.pair)
-                if order['status'] == 'closed':
-                    trade.sell_reason = SellType.STOPLOSS_ON_EXCHANGE.value
-                    trade.update(order)
-                    self.notify_sell(trade)
-                    result = True
-                elif self.config.get('trailing_stop', False):
-                    # if trailing stoploss is enabled we check if stoploss value has changed
-                    # in which case we cancel stoploss order and put another one with new
-                    # value immediately
-                    self.handle_trailing_stoploss_on_exchange(trade, order)
-        except DependencyException as exception:
-            logger.warning('Unable to create stoploss order: %s', exception)
-        return result
+            except DependencyException as exception:
+                logger.warning('Unable to place a stoploss order on exchange: %s', exception)
+
+        # If stoploss order is canceled for some reason we add it
```

Comment:
This behaviour must be documented ... it may leave users trying to remove a stoploss on the exchange, and the bot adding it again and again.

Reply:
Disagree, bot positions and stakes should not be touched manually, otherwise we cannot guarantee anything. but nevertheless where do you want to document it ?

Reply:
i agree, it should not be touched - but you can't prevent it, and it's not written anywhere.

i would add it as a note/warning on the documentation for "stoploss_on_exchange" - in `configuration.md` probably

Reply:
don't you think we have to include it somewhere on the main page? cause I think this is general, you don't run the bot on the account and trying to have fun with its positions ...